spec/controllers/sign_up/recovery_codes_controller_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe SignUp::RecoveryCodesController do
+  describe '#update' do
+    it 'redirects to the profile page' do
+      stub_sign_in
+      subject.current_user.recovery_code = 'foo'
+
+      patch :update
+
+      expect(response).to redirect_to profile_path
+    end
+  end
+end
```

Comment:
Added controller spec at d1ade8c]

Note that I kept one feature spec because I realized I wanted to make sure that our other controller was properly setting the session var, which required clicking through the pages